### ISSUE TITLE
systemd-networkd: Fails to handle local router advertisements

### ISSUE THREAD

**Issue by wak-google** [systemd/systemd#18963]
**systemd version the issue has been seen with**
v245

**Used distribution**
Yocto / OpenBMC

**Linux kernel version used** (`uname -a`)
Linux bmc 5.4.99-nuvoton-dirty-48923e0 #1 SMP Thu Mar 11 02:14:29 UTC 2021 armv7l GNU/Linux

**CPU architecture issue was seen on**
armv7

**Expected behaviour you didn't see**
systemd-networkd should have ignored the RA

**Unexpected behaviour you saw**
systemd-networkd shut down all services on the link

**Steps to reproduce the problem**
Enable outbound router advertisement and accept them on the same link. Send an RS from the same machine to systemd-networkd, then watch it fail when it processes the RA from itself.

**Additional program output to the terminal or log subsystem illustrating the issue**
```
tcpdump: listening on mgmt, link-type EN10MB (Ethernet), capture size 262144 bytes
00:35:39.726456 IP6 (flowlabel 0x69aea, hlim 255, next-header ICMPv6 (58) payload length: 88) fe80::a0d5:9dff:fe6c:6f01 > ff02::1: [icmp6 sum ok] ICMP6, router advertisement, length 88
	hop limit 0, Flags [none], pref medium, router lifetime 60s, reachable time 0ms, retrans timer 0ms
	  source link-address option (1), length 8 (1): b2:18:14:5e:ec:01
	  mtu option (5), length 8 (1):  1500
	  prefix info option (3), length 32 (4): 2620:15c:2c5:a954:fd::/80, Flags [onlink, auto], valid time 60s, pref. time 60s
	  route info option (24), length 24 (3):  2620:15c:2c5:a954:fd01::/80, pref=medium, lifetime=60s
00:35:40.304155 IP6 (flowlabel 0x62624, hlim 255, next-header ICMPv6 (58) payload length: 16) fe80::a0d5:9dff:fe6c:6f01 > ff02::2: [icmp6 sum ok] ICMP6, router solicitation, length 16
	  source link-address option (1), length 8 (1): b2:18:14:5e:ec:01
00:35:42.324397 IP6 (flowlabel 0x62624, hlim 255, next-header ICMPv6 (58) payload length: 8) fe80::a0d5:9dff:fe6c:6f01 > ff02::2: [icmp6 sum ok] ICMP6, router solicitation, length 8
00:35:42.326109 IP6 (flowlabel 0x69aea, hlim 255, next-header ICMPv6 (58) payload length: 88) fe80::a0d5:9dff:fe6c:6f01 > ff02::1: [icmp6 sum ok] ICMP6, router advertisement, length 88
	hop limit 0, Flags [none], pref medium, router lifetime 0s, reachable time 0ms, retrans timer 0ms
	  source link-address option (1), length 8 (1): b2:18:14:5e:ec:01
	  mtu option (5), length 8 (1):  1500
	  prefix info option (3), length 32 (4): 2620:15c:2c5:a954:fd::/80, Flags [onlink, auto], valid time 60s, pref. time 60s
	  route info option (24), length 24 (3):  2620:15c:2c5:a954:fd01::/80, pref=medium, lifetime=60s
```

```
Jan 01 00:35:39 systemd[1]: Started Network Service.
Jan 01 00:35:39 systemd-networkd[1490]: mgmtdummy: netdev exists, using existing without changing its parameters
Jan 01 00:35:39 systemd-networkd[1490]: mgmt: netdev exists, using existing without changing its parameters
Jan 01 00:35:39 systemd-networkd[1490]: mgmt: IPv6 successfully enabled
Jan 01 00:35:39 systemd-networkd[1490]: eth0: IPv6 successfully enabled
Jan 01 00:35:42 systemd-networkd[1490]: mgmt: Could not set NDisc route or address: Gateway can not be a local address. Invalid argument
Jan 01 00:35:42 systemd-networkd[1490]: mgmt: Failed
Jan 01 00:36:04 systemd-networkd[1490]: eth0: Could not set neighbor, ignoring: Connection timed out
```

**Comment by wak-google** [comment]
I looked at the code around src/network/networkd-ndisc.c, and it looks like this actually should be fixed in 221019166f315252304b3459902ead613b905de5. Need to verify this fixes it for me.

**Comment by wak-google** [comment]
Verified that this is fixed for me using that commit
